core/ecschema-editing/src/test/Merging/ClassMergerOrder.test.ts

```diff
@@ -0,0 +1,398 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import {  EntityClass, Mixin, Schema, SchemaContext, StructClass } from "@itwin/ecschema-metadata";
+import { SchemaMerger } from "../../ecschema-editing";
+import { expect } from "chai";
+
+describe("Class items merging order tests", () => {
+  let context: SchemaContext;
+
+  const sourceJson = {
+    $schema: "https://dev.bentley.com/json_schemas/ec/32/ecschema",
+    name: "SourceSchema",
+    version: "1.2.3",
+    alias: "source",
+  };
+
+  const targetJson = {
+    $schema: "https://dev.bentley.com/json_schemas/ec/32/ecschema",
+    name: "TargetSchema",
+    version: "1.0.0",
+    alias: "target",
+  };
+
+  beforeEach(async () => {
+    context = new SchemaContext();
+  });
+
+  describe("Entity and Mixing class merging order tests", () => {
+    it("should merge the missing entity class with derived base class before the base class", async () => {
```

Comment:
'derived base class' doesn't make sense, just 'base class' seems to make sense for this tests case